### PR TITLE
STUD-87- add task front end 1st

### PR DESCRIPTION
**Pull request by AhmedElemary57**
** Add task front end and some connection between FE and BE **
1. add task front end version one the next version should demonstrate the form in a popup window also the date should be fixed.
2. Get all tasks related to a student.
3. trigger the task from todo to done and from done to todo. 
4. remove the task permanently.
5. The connection from back to front was done for the above properties 2,3,4.  
We may need to run .. ". npm install --save ngx-material-timepicker " for add task page

### CODE REVIEW

File: Student-Unity-frontend/src/app/components/task/task.component.ts [hunk 1]
```diff
@@ -11,39 +11,47 @@ import { STATUS } from 'src/app/models/Status';
...
+  @Output() markAsDoneEvent = new EventEmitter<Task>();
+  @Output() unMarkAsDoneEvent = new EventEmitter<Task>();
 
   constructor(private studentTaskService: StudentTaskService) { }
 
   ngOnInit(): void {
   }
 
   public removeTask() {
+    this.removingDoneTaskEvent.emit(this.task.taskId);
```
Comment (Ahmedelsa3eed) [inline Student-Unity-frontend/src/app/components/task/task.component.ts:25]: i think you forgot to delete this line, right?

File: Student-Unity-frontend/src/app/components/task/task.component.ts [hunk 2]
```diff
@@ -11,39 +11,47 @@ import { STATUS } from 'src/app/models/Status';
...
-          else {
-            this.task.status = STATUS.DONE;
-            this.markAsDoneEvent.emit(this.task.taskId);
+        console.log(res);
+        console.log("task", this.task);
+        if (res.status === 200) {
+          console.log("in if condition");
+          if (this.task.status === true) {
+            console.log("unmark as done");
+            this.unMarkAsDoneEvent.emit(this.task);
```
Comment (Ahmedelsa3eed) [inline Student-Unity-frontend/src/app/components/task/task.component.ts:47]: you should set the status before you emit the task

File: Student-Unity/src/main/java/csed/swe/studentunity/DAO/StudentTaskRepo.java [hunk 3]
```diff
@@ -10,32 +10,33 @@
 
 public interface StudentTaskRepo extends JpaRepository<StudentTask, StudentTaskId> {
 
-    @Query("SELECT new Task (t.title, t.dueDate, t.telegramLink), new StudentTask(st.status) " +
-            "FROM Task AS t, StudentTask AS st, User AS u " +
-            "WHERE t = st.studentTaskId.task AND u.id = :userId AND st.studentTaskId.user.id = :userId"
+    @Query(value="SELECT t.task_id,t.course_code,t.title,t.due_date,t.telegram_link,st.status\n" +
+            "FROM task AS t, student_task AS st, user AS u \n" +
```
Comment (Ahmedelsa3eed) [inline Student-Unity/src/main/java/csed/swe/studentunity/DAO/StudentTaskRepo.java:14]: you should avoid cross join

File: Student-Unity-frontend/src/app/components/task/task.component.ts (hunk 1 above)
Comment (AhmedElemary57) [reply Student-Unity-frontend/src/app/components/task/task.component.ts:25]: Yes I forgot to do so

File: Student-Unity/src/main/java/csed/swe/studentunity/DAO/StudentTaskRepo.java (hunk 3 above)
Comment (AhmedElemary57) [reply Student-Unity/src/main/java/csed/swe/studentunity/DAO/StudentTaskRepo.java:14]: Why ? 
I think that it is the same If a WHERE clause is added to a cross join, it behaves as an inner join as the WHERE imposes a limiting factor.

I do not know why you said that but to my knowledge the query optimizer do that for you.

You may be right I don't know we can ask ENG El-Naggar about that.
I used the join in that way as I see it more readable

File: Student-Unity-frontend/src/app/components/task/task.component.ts (hunk 2 above)
Comment (AhmedElemary57) [reply Student-Unity-frontend/src/app/components/task/task.component.ts:47]: Yes it is more reasonable but it is the same as I do not check in the parent component about the status I just but the child component in the observable array l

File: Student-Unity/src/main/java/csed/swe/studentunity/DAO/StudentTaskRepo.java (hunk 3 above)
Comment (Ahmedelsa3eed) [reply Student-Unity/src/main/java/csed/swe/studentunity/DAO/StudentTaskRepo.java:14]: ok we can discuss it later

**Review by Ahmedelsa3eed — APPROVED** [review]

**Pull request merged**